semgrep-core/src/fixing/Autofix.ml

```diff
@@ -94,6 +94,25 @@ let apply_edits_to_text text edits =
   if discarded_edits = [] then Success fixed_text
   else Overlap { partial_result = fixed_text; discarded_edits }
 
+let partition_edits_by_file edits =
+  let edits_by_file = Hashtbl.create 8 in
+  List.iter
```

Comment:
probably can use one of the Common.group_by function to do this.

Reply:
Looks like `Common.group_by` reverses the order of the list elements. The order is important here. If I reverse it after calling `group_by`, then I'm relying on `group_by` to always reverse the order of the list elements. If someone changes it later, this will break.

```
let group_by f xs =
  (* use Hashtbl.find_all property *)
  let h = Hashtbl.create 101 in

  (* could use Set *)
  let hkeys = Hashtbl.create 101 in

  xs |> List.iter (fun x ->
    let k = f x in
    Hashtbl.replace hkeys k true;
    Hashtbl.add h k x
  );
  Hashtbl.fold (fun k _ acc -> (k, Hashtbl.find_all h k)::acc) hkeys []
;;

let l = ["1"; "2"; "3"];;

let g = group_by (fun _ -> 5) l;;

(* output: 3, 2, 1 *)
print_endline (g |> List.hd |> snd |> String.concat ", ");;
```